test/support/test_config.h

```diff
@@ -66,13 +66,15 @@
 #define TEST_TBB_BACKEND_PRESENT (!defined(ONEDPL_USE_TBB_BACKEND) || ONEDPL_USE_TBB_BACKEND != 0)
 
 // Check for C++ standard and standard library for the use of ranges API
-#define _TEST_RANGES_FOR_CPP_17_DPCPP_BE_ONLY (__cplusplus >= 201703L && TEST_DPCPP_BACKEND_PRESENT)
+#if !defined(_ENABLE_RANGES_TESTING)
+#define _TEST_RANGES_FOR_CPP_17_DPCPP_BE_ONLY (__cplusplus >= 201703L && _ONEDPL_BACKEND_SYCL)
```

Comment:
I believe we should not use  `_ONEDPL_BACKEND_SYCL` macro here. If we do so, then we must include oneDPL headers before `test_config.h`. Some tests include `test_config.h` first. Let's use `TEST_DPCPP_BACKEND_PRESENT` instead.